bazel/python_utils/coverage_decorator.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2022 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Inspired by https://github.com/bazelbuild/bazel/issues/10660#issuecomment-924223507 by @joshua-cannon-techlabs
+# Contributed to https://github.com/bazelbuild/bazel under Apache-2.0
+
+"""
+Decorator used for generating test coverage with Bazel.
+"""
+
+import contextlib
+import os
+import pathlib
+from typing import List
+
+import coverage
+from coverage_lcov import converter
+
+# environment variable COVERAGE - should be 1 iff bazel coverage was used
+COVERAGE_ENV = "COVERAGE"
+# environment variable COVERAGE_DIR - bazel sandbox folder where coverage of traget is created
+COVERAGE_DIR_ENV = "COVERAGE_DIR"
+# environment variable COVERAGE_MANIFEST - bazel instrumented files, created via --instrumentation_filter
+COVERAGE_MANIFEST_ENV = "COVERAGE_MANIFEST"
+# environment variable COVERAGE_OUTPUT_FILE - location of generated lcov file
+COVERAGE_OUTPUT_FILE_ENV = "COVERAGE_OUTPUT_FILE"
```

Comment:
Where are these environment variables set? In what scenarios would we change these values?

Reply:
@themarwhal Made this more clear in documentation. In short: environment variables are set by Bazel when 'bazel coverage' is executed. I do not see a scenario in our setup where we would change the values. They are currently read in code because this is a workaround - that should be reverted when bazel supports coverage for Python ootb.